src/librustc/ty/layout.rs

```diff
@@ -1558,10 +1557,22 @@ impl<'a, 'tcx> LayoutDetails {
                 let mut start_align = Align::from_bytes(256, 256).unwrap();
                 assert_eq!(Integer::for_abi_align(dl, start_align), None);
 
+                // repr(C) on an enum tells us to make a (tag, union) layout,
+                // so we need to compute the alignment of the union here.
+                // (enum_union_align is checked by univariant_uninterned)
+                let mut union_align = min_ity.align(dl);
```

Comment:
Rename this to `prefix_align`, I guess.

Reply:
But it's not how aligned the prefix should be, it's how aligned the *suffix* should be.

Reply:
It's the alignment imposed by the prefix, the logic looking at it *need not know* why it's chosen the way it is.